.gitlab-ci.yml

```diff
@@ -325,6 +325,23 @@ pkg:nix:deploy:
     - master
     - /^v.*\..*$/
 
+pkg:nix:deploy:channel:
+  stage: deploy
+  environment:
+    name: cachix
+    url: https://coq.cachix.org
+  dependencies:
+    - pkg:nix:deploy
+  before_script:
+    - which ssh-agent || ( apt-get update -y && apt-get install openssh-client -y )
+    - eval $(ssh-agent -s)
+    - echo "$GITHUB_DEPLOY_KEY" | tr -d '\r' | ssh-add - > /dev/null
+    - mkdir -p ~/.ssh
+    - chmod 700 ~/.ssh
+    - ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+  script:
+    - git push git@github.com:coq/coq "${CI_COMMIT_REF_NAME}-cachix"
```

Comment:
The SSH key should not allow to write to the central repository. I recommend to use a different repository (as you mentioned): if said repository get compromised, that would not be too serious.

Reply:
Note that the repository has an additional security in the form of protected branches (to which no one can force-push and non-admins can only push approved PRs).

Reply:
I'd be very concerned if we go this route, in particular given the recent problems we've seen... Can you explain for non-experts why this info needs to be pushed to the Coq repo?

Reply:
There is no strong need. I am OK with the suggestion of @vbgl to put this in a separate repo.